pkg/feature/feature.go

```diff
@@ -71,35 +74,32 @@ func (f *Feature) Apply() error {
 		return multiErr.ErrorOrNil()
 	}
 
-	// create or update resources
+	// Create or update resources
 	for _, resource := range f.resources {
 		if err := resource(f); err != nil {
 			return err
 		}
 	}
 
 	// Process and apply manifests
-	for _, m := range f.manifests {
-		if err := m.processTemplate(f.Spec); err != nil {
+	for i, m := range f.manifests {
+		if err := m.process(f.fsys, f.Spec); err != nil {
 			return errors.WithStack(err)
 		}
 
-		log.Info("converted template to manifest", "feature", f.Name, "path", m.targetPath())
+		f.manifests[i] = m
```

Comment:
why is this needed?

Reply:
since range() creates a copy of each element in the slice and we modify the copied manifest, we update the original slice with the modified manifest.

Reply:
Can we do `f.manifests[i].process` within the `if` rather than processing in the copy?

Reply:
I think we can simply do 

```golang
for i := range f.manifests {
	if err := f.manifests[i].process(f.Spec); err != nil {
		return errors.WithStack(err)
	}
}
```

Reply:
agreed, thx! 
3c6faeb